SuperHero/src/test/java/ApiTests.java

```diff
@@ -0,0 +1,280 @@
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import io.restassured.response.ValidatableResponse;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.testng.annotations.BeforeClass;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static io.restassured.RestAssured.given;
+public class ApiTests {
+	private String baseUrl = "https://superhero.qa-test.csssr.com/superheroes";
+	@BeforeClass
+	public void setUp() {
+		RestAssured.baseURI = baseUrl;
+	}
+	@Test
+	@DisplayName("Add superhero with all required fields")
+	void postTest() {
+		String body = """
+				    {
+				        "birthDate": "2019-02-21",
+				        "city": "GOTHAM",
+				        "fullName": "BATMAN",
+				        "gender": "M",
+				        "mainSkill": "fight"
+				    }
+				""";
+		ValidatableResponse response =
+				given().
+						header("accept", "application/json").
+						header("Content-Type", "application/json").
+						body(body).
+						when().
+						post(baseUrl).then();
+		Response response1 = response.extract().response();
+		int status = response1.statusCode();
+		String responseBody = response1.body().prettyPrint();
+		Assertions.assertEquals(200, status, "Статус код не соответствует ожидаемому");
+		String actualResponseBody = response1.body().asString();
+		String id = response1.jsonPath().getString("id");
+		Assertions.assertNotNull(id, "id не должен быть null");
+		String actualBirthDate = response1.jsonPath().getString("birthDate");
+		assertEquals("2019-02-21", actualBirthDate, "birthDate не соответствует ожидаемому значению");
+
+		String actualCity = response1.jsonPath().getString("city");
+		assertEquals("GOTHAM", actualCity, "Город не соответствует ожидаемому");
+
+		String actualFullName = response1.jsonPath().getString("fullName");
+		assertEquals("BATMAN", actualFullName, "Полное имя не соответствует ожидаемому");
+
+		String actualGender = response1.jsonPath().getString("gender");
+		assertEquals("M", actualGender, "Пол не соответствует ожидаемому");
+
+		String actualMainSkill = response1.jsonPath().getString("mainSkill");
```

Comment:
Since you have several assertions on the same json it could be simplified by using several ways. We will go into it deeper separately or you can check by yourself topics:
1) SoftAssertions
2) Parsing response into Java object
3) Using JUnit to compare to objects (it even possible to ignore some fields like Id, or date)